### ISSUE TITLE
[Quota] App- or System-Quota not applied when configured only one

### ISSUE THREAD

**Issue by cbrowet-axway** [Axway-API-Management-Plus/apimanager-swagger-promote#55]
Looks like application quota's are not applied.
I see `Property: applicationQuota has no handler configured and is not a propertyHandler` in the log.
This is with 1.5.1

Log
```
2103   INFO        APIChangeState| Changed property: image[Desired: 'APIImage [filename=./samples/images/sample-api-icon.jpg]' vs Actual: 'APIImage [filename=/api/portal/v1.3/discovery/swagger/apis/97e4b1ae-cc07-4961-a3f3-2db13bdbcd08/image]']
2104   INFO        APIChangeState| Changed property: applicationQuota[Desired: 'APIQuota [id=null, type=APPLICATION, restrictions=[QuotaRestriction [api=null, method=*, type=throttle, config={period=seconds, messages=1, per=10}]]]' vs Actual: 'null']
2104   INFO     APIManagerAdapter| Strategy: Going to update existing API: CalculatorRestTEST (Version: 3.0)
2105   INFO     APIManagerAdapter| Recognized the following changes. Potentially Breaking: [] plus Non-Breaking: [image, applicationQuota]
2105   INFO     APIManagerAdapter| Strategy: Update existing API, as all changes can be applied in current state.
2107   DEBUG    APIManagerAdapter| Updating API-Proxy
2127   DEBUG    APIManagerAdapter| Going to update property: image
2128   DEBUG    APIManagerAdapter| Property: image has no handler configured and is not a propertyHandler
2128   DEBUG    APIManagerAdapter| Going to update property: applicationQuota
2128   DEBUG    APIManagerAdapter| Property: applicationQuota has no handler configured and is not a propertyHandler
2215   INFO     APIManagerAdapter| Updating API-Image from: ./samples/images/sample-api-icon.jpg
2275   INFO     APIManagerAdapter| All desired applications: [[Testing]] have already a subscription. Nothing to do.
2275   INFO                   App| Successfully replicated API-State into API-Manager
```

Config:
```
{
   "name":"CalculatorRestTEST",
   "path":"/api/v1/calculator/rest",
   "state":"published",
   "summary":"My complete API-Summary",
   "version":"3.0",
   "organization":"Test",
   "applications":[
      { "name":"Testing" }
   ],
   "descriptionType":"manual",
   "descriptionManual":"This is my __markdown__ based API description.",
   "vhost":"server.semperpax.com:8067",
   "backendBasepath":"http://localhost:8081",
   "image": "./samples/images/sample-api-icon.jpg", 
   "securityProfiles":[
      {
         "name":"_default",
         "isDefault":true,
         "devices":[
            {
               "type":"authPolicy",
               "name":"Invoke Policy",
               "order":1,
               "properties":{
                  "authenticationPolicy":"Generic Inbound Policy",
                  "useClientRegistry":false,
                  "subjectSelector":"authentication.subject.id",
                  "descriptionType":"original",
                  "descriptionUrl":"",
                  "descriptionMarkdown":"",
                  "description":""
               }
            }
         ]
      }
   ],
   "outboundProfiles":{
      "_default":{
         "routePolicy":"CalculatorFromRest"
      }
   },
   "applicationQuota":{
      "restrictions":[
         {
            "method":"*",
            "type":"throttle",
            "config":{
               "period":"seconds",
               "messages":1,
               "per":10
            }
         }
      ]
   }
}
```

**Comment by cwiechmann** [comment]
@cbrowet-axway, ah, you are right! I haven't an integration test only having App- or System-Quota and therefore haven't realized this bug in UpdateQuotaConfiguration.java:
```
if(desiredState.getApplicationQuota()==null || desiredState.getSystemQuota()==null) return;
```
You see, I used an OR condition and therefore your Quota is silently ignored.

I'll add an integration-test for that scenario and fix this stupid bug. Thanks for reporting this.

**Comment by cbrowet-axway** [comment]
Thanks 👍

**Comment by cbrowet-axway** [comment]
BTW, the wiki mentions "period: Must be either [seconds|minutes|**hours**|days|weeks]", while it's "hour" if not mistaken?
Unfortunately, the manager api swagger does not describe the parameter details, either.

Might be useful to say that "throttlemb" implies "mb" while "throttle" implies "messages", too ;)

**Comment by cwiechmann** [comment]
Thanks again! Already fixed/improved the documentation.

**Comment by cwiechmann** [comment]
@cbrowet-axway, I added the new integration-test: AppOrSystemQuotaOnlyTestIT and made the bugfix. 

@cbrowet-axway a question: Suppose you have done the following:
1. Configure Application-Quota for your API
2. Promote it and App-Quota is set
3. You change your mind, remove the App-Quota config and configure a system quota only
4. System quota is applied, BUT also the Application-Quota stays as before!

Would that be an acceptable behavior for you or should the existing App-Quota removed?

**Comment by cbrowet-axway** [comment]
I'd say it should be WYSIWYG, so the config should exactly represent the final state, otherwise it will be confusing.
So, ideally, remove the quota if not in the config.

I went ahead and built with the change and, strangely, although all seems fine on promote side, the quota is not reflected in the webui and not applied (on 7.7).
Log shows
```
2213   DEBUG       APIChangeState| No change for property: applicationQuota[Desired: 'APIQuota [id=null, type=APPLICATION, restrictions=[QuotaRestriction [api=null, method=*, type=throttle, config={period=seconds, messages=1, per=10}]]]' vs Actual: 'APIQuota [id=null, type=APPLICATION, restrictions=[QuotaRestriction [api=97e4b1ae-cc07-4961-a3f3-2db13bdbcd08, method=*, type=throttle, config={messages=1, period=seconds, per=10}]]]']
```
so seems that the manager actually has the quota. Manager bug?
Does it work for you and on which version?


https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/commit/7f354380bbcceca0a0430b66495cfe04941e6762#diff-485bdb829bbeba52e1267dec8a937e7aR63 Wrong logging ;)

**Comment by cbrowet-axway** [comment]
"desiredState.getSystemQuota" -> "desiredState.get**Application**Quota" is what I had in mind ;)

**Comment by cbrowet-axway** [comment]
Ah, seems the quota is set at "Application defaults" level, rather than at the level of the application(s) related to the api.
Is it by design? In that case, I'd open a feature request ;)

As the application already had specific quotas, the default is ignored...


![image](https://user-images.githubusercontent.com/50665603/57848691-cb10df80-77d9-11e9-9c4c-94d64873ab9c.png)

**Comment by cwiechmann** [comment]
Yes, this is by design as you control the API and not the application itself. I think such feature would be part of another tool that controls applications. 
FYI: Swagger-Promote (with the next version) takes over manually configured applications quotes. See here #51

**Comment by cwiechmann** [comment]
@cbrowet-axway, concerning WYSIWG: Having it not declared in the desired API state is difficult as you haven't declared anything. Hope it's clear what I mean with that.
Right now, all properties not specifically declared, are ignored. I made this decision for performance reasons, as it would mean to compare really everything with that actual state and this leads to even more REST-API-Calls. The tool is already doing a lot REST-API-Calls.

Perhaps adding a generic value like: NONE which could be used for any kind of properties to clearly say: I don't want this. This would be another issue.

**Comment by cbrowet-axway** [comment]
Yep, an explicit NONE seems the proper way to me

**Comment by cwiechmann** [comment]
Has been released with version 1.5.2.
